doc/rados/configuration/bluestore-config-ref.rst

```diff
@@ -233,66 +248,65 @@ The data fraction can be calculated by
-checksumming is handled by RocksDB and uses `crc32c`. Data
-checksumming is done by BlueStore and can make use of `crc32c`,
-`xxhash32`, or `xxhash64`.  The default is `crc32c` and should be
-suitable for most purposes.
-
-Full data checksumming does increase the amount of metadata that
-BlueStore must store and manage.  When possible, e.g., when clients
-hint that data is written and read sequentially, BlueStore will
-checksum larger blocks, but in many cases it must store a checksum
-value (usually 4 bytes) for every 4 kilobyte block of data.
-
-It is possible to use a smaller checksum value by truncating the
-checksum to two or one byte, reducing the metadata overhead.  The
-trade-off is that the probability that a random error will not be
-detected is higher with a smaller checksum, going from about one in
-four billion with a 32-bit (4 byte) checksum to one in 65,536 for a
-16-bit (2 byte) checksum or one in 256 for an 8-bit (1 byte) checksum.
-The smaller checksum values can be used by selecting `crc32c_16` or
-`crc32c_8` as the checksum algorithm.
-
-The *checksum algorithm* can be set either via a per-pool
-``csum_type`` property or the global config option.  For example:
+BlueStore checksums all metadata and all data written to disk. Metadata
+checksumming is handled by RocksDB and uses the `crc32c` algorithm. By
+contrast, data checksumming is handled by BlueStore and can use either
+`crc32c`, `xxhash32`, or `xxhash64`. Nonetheless, `crc32c` is the default
+checksum algorithm and it is suitable for most purposes.
+
+Full data checksumming increases the amount of metadata that BlueStore must
+store and manage. Whenever possible (for example, when clients hint that data
+is written and read sequentially), BlueStore will checksum larger blocks. In
+many cases, however, it must store a checksum value (usually 4 bytes) for every
+4 KB block of data.
+
+It is possible to obtain a smaller checksum value by truncating the checksum to
+one or two bytes and reducing the metadata overhead.  A drawback of this
+approach is that it increases the probability of a random error going
+undetected: about one in four billion given a 32-bit (4 byte) checksum, 1 in
+65,536 given a 16-bit (2 byte) checksum, and 1 in 256 given an 8-bit (1 byte)
+checksum. To use the smaller checksum values, select `crc32c_16` or `crc32c_8`
+as the checksum algorithm.
+
+The *checksum algorithm* can be specified either via a per-pool ``csum_type``
+configuration option or via the global configuration option. For example:
 
 .. prompt:: bash $
 
    ceph osd pool set <pool-name> csum_type <algorithm>
 
 .. confval:: bluestore_csum_type
 
-Inline Compression
+inline compression
 ==================
 
-BlueStore supports inline compression using `snappy`, `zlib`, or
-`lz4`. Please note that the `lz4` compression plugin is not
+bluestore supports inline compression using `snappy`, `zlib`, or
+`lz4`. please note that the `lz4` compression plugin is not
```

Comment:
HUH? News to me, I'm using it right now.  Maybe this was once true.  Curiously `test/compressor/test_compression.cc` appears to not test `zstd`.

>BlueStore offers optional inline compression using `snappy`, `zlib`, `zstd`, or`lz4`.

Reply:
This was a change introduced on 02 Aug 2017. Given that (1) that is nearly six years ago, and (2) in this domain, you are the boots-on-the-ground non-plus-ultra dealer of in-the-field expertise, we are going to remove the note that says that the ``lz4`` compression plugin is not supported. But we're going to remove the note in https://github.com/ceph/ceph/pull/51787, to keep things tidier and easier for me.

Reply:
Sure.  Be sure to add `zstd` there as well.

Reply:
<greg hirsch>If it is to be said, then let it be so.<\greg hirsch>

Reply:
<IronMaiden>So it shall be written; so it shall be done</IronMaiden>
<Mr. Wizard>Be what you is, and not what you is not.  Folks what do this is the happiest lot</Mr. Wizard>